org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/util/ResourceLoadHelper.java

```diff
@@ -16,7 +16,9 @@
  * Abstraction for other test utilities that want to load a resource for a
  * textual model.
  * @author Sebastian Zarnekow - Initial contribution and API
+ * @deprecated Use org.eclipse.xtext.ui.testing.util.ResourceLoadHelper instead
  */
+@Deprecated
```

Comment:
I don't think deprecating this class is providing added value.

Reply:
Since the entire `org.eclipse.xtext.junit4` project is deprecated (has been superseded by the `org.eclipse.xtext.testing` project), I think it is worth marking all elements (classes/interfaces) in that project as deprecated (where there is a corresponding alternative to use) to see where in the Xtext code-base deprecated elements are used and where there is a need for restructuring to eliminate the usage of deprecated elements from the Xtext code-base.

Reply:
I can live with this, if (and only if) we eliminate most of the usages of the now-deprectated code from the Xtext codebase. In this case, this is trivial to show that the "new" solution is working at least as good as the old one, but generally speaking this is to ensure that the new code fulfils the same requirements as the old code.
The drawback of that is, that the now-deprecated code is then subject to be forgotten since it would no longer be tested. Interesting twist, I suppose.

If you want to work on the deprecation of the old junit4 bundles, I recommend to do it completely or not at all. E.g. add a package-info to all the packages in the deprecated bundle, and mark the entire package as deprecated with a hint that points to its new equivalent. There is no reason to make ResourceLoadHelper deprecated but keep AbstractXtextTests as non-deprecated base type. Before that, we'd need to figure a migration path for the tests, though. If AbstractXtextTests stays as is, the entire deprecation effort is as half-hearted as it is right now. And if it is deprecated, we'll have a lot more warnings in the code base.

Reply:
I am fine with that, I created an [issue](https://github.com/eclipse/xtext-eclipse/issues/1430) for that work. I am closing this PR without merging it.